### PR TITLE
chore: relinted, updated

### PR DESCRIPTION
**Pull request by fredbi**
* updated go version to go1.19
* updated linting rules
* updated dependencies, upgraded yaml.v2 to yaml.v3
* relinted code
* doc: updated links and badges in README
* ci: updated github actions

### CODE REVIEW

File: result.go [hunk 1]
```diff
@@ -132,7 +132,8 @@ func (r *Result) RootObjectSchemata() []*spec.Schema {
 }
 
 // FieldSchemata returns the schemata which apply to fields in objects.
-// nolint: dupl
+//
+
```
Comment (youyuanwu) [inline result.go:136]: Extra empty line?

File: result.go [hunk 2]
```diff
@@ -152,7 +153,8 @@ func (r *Result) FieldSchemata() map[FieldKey][]*spec.Schema {
 }
 
 // ItemSchemata returns the schemata which apply to items in slices.
-// nolint: dupl
+//
+
```
Comment (youyuanwu) [inline result.go:157]: Extra empty line?

**Review by youyuanwu — APPROVED** [review]

File: result.go (hunk 1 above)
Comment (fredbi) [reply result.go:136]: yes we do, normally. Let me remove that extra empty line.

File: result.go (hunk 2 above)
Comment (fredbi) [reply result.go]: @youyuanwu fixed

**Pull request merged**